### PR TITLE
Persist settings across relaunch: Aspect ratio options

### PR DESCRIPTION
**Pull request by karaggeorge**
This PR:
1. Adds this feature: https://github.com/wulkano/kap/issues/219
2. Solves the ratio issue (I don't know if it's documented or not, but ratios would show up as crazy long numbers)

Worked on each doing the following:
1. Use the `settings-manager` to save and load the values of the `dimensions` object which includes:
    - width
    - height
    - ratio
    - ratioLocked
2. Added a `Math.round` call before calling the `getSimplestRatio` function since `LargestCommonDivisor` as a mathematical concept only works with integers. The display values were rounded but the actual values in the `dimensions` object were not, which was causing the long decimals

### CODE REVIEW

File: app/src/renderer/js/main.js [hunk 1]
```diff
@@ -40,23 +40,27 @@ document.addEventListener('DOMContentLoaded', () => {
...
 
   // Init dynamic elements
   if (app.kap.settings.get('recordAudio') === true) {
     toggleAudioRecordBtn.classList.add('is-active');
     micOnIcon.classList.remove('hidden');
     micOffIcon.classList.add('hidden');
   }
+    
+  // Set initial values
+  [inputWidth.value, inputHeight.value] = [width, height].map(Math.round);
```
Comment (albinekb) [inline app/src/renderer/js/main.js]: Haha clever 😍

Comment (sindresorhus) [reply app/src/renderer/js/main.js]: I think it's too clever. This is much more readable and only 4 characters more:

```js
inputWidth.value = Math.round(width);
inputHeight.value = Math.round(height);
```

File: app/src/common/settings-manager.js [hunk 2]
```diff
@@ -14,7 +14,13 @@ const DEFAULTS = {
   highlightClicks: false,
   fps: 30,
   recordAudio: false,
-  audioInputDeviceId: 'none'
+  audioInputDeviceId: 'none',
+  dimensions: {
+    height: 512,
+    width: 512,
+    ratio: '1:1',
```
Comment (sindresorhus) [inline app/src/common/settings-manager.js]: I think we should persist it as plain numbers. I know it's already represented as a string in the code, but the code we can change later. It's much harder to change the settings.

So it should be saved as `ratio: [1, 1]`

Comment (karaggeorge) [reply app/src/common/settings-manager.js]: I can do that. Shouldn't be too big of a change. Do you want it as part of this PR?

File: app/src/renderer/js/main.js (hunk 1 above)
Comment (karaggeorge) [reply app/src/renderer/js/main.js]: Do you want me to change it?

File: app/src/common/settings-manager.js (hunk 2 above)
Comment (sindresorhus) [reply app/src/common/settings-manager.js]: Sure

File: app/src/renderer/js/main.js (hunk 1 above)
Comment (sindresorhus) [reply app/src/renderer/js/main.js]: Yes, please :)

**Comment by sindresorhus** [comment]
And some linter issues that need to be fixed: https://circleci.com/gh/wulkano/kap/454?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

File: app/src/renderer/js/main.js (hunk 1 above)
Comment (karaggeorge) [reply app/src/renderer/js/main.js]: 0fbd884

File: app/src/common/settings-manager.js (hunk 2 above)
Comment (karaggeorge) [reply app/src/common/settings-manager.js]: 600a667

**Comment by karaggeorge** [comment]
- Fixed the linting issues: 0fbd884
- Save ratio as an array: 600a667

In order to save the ratio as an array, I had to modify the way setSelectedRatio works a bit, otherwise, you would see a different (re-calculated) ratio when you opened the app again

**Comment by sindresorhus** [comment]
There's a lot of `app.kap.settings.set('dimensions', dimensions);` calls. I made [this module](https://github.com/sindresorhus/on-change) today that you could use to remove all the duplicate calls and put it in one place. Would you be willing to use it in your PR? Otherwise I'll do a follow-up commit, no worries.

```js
const dimensions = onChange(app.kap.settings.get('dimensions'), () => {
  app.kap.settings.set('dimensions', dimensions);
});

**Comment by karaggeorge** [comment]
@sindresorhus That's a quite interesting package. I like it. However when adding it to this project, I got the following issue:

When this gets called `app.kap.settings.set('dimensions', dimensions);` I get the following error:
screen shot 2017-12-11 at 12 43 45

I think it has something to do with what this:`settings.setSync(key, value)` calls on the value object internally

Any thoughts?

**Comment by sindresorhus** [comment]
@karaggeorge I need to dig into it later. Shouldn't block this change though.

**Comment by karaggeorge** [comment]
@sindresorhus then it should be good to go once it's reviewed. I don't think there's anything else to add

**Comment by matheuss** [comment]
Tested, very nice! Changed the tittle since I see nothing related to export format in the diff 🤔

**Pull request merged**